packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -62,7 +65,23 @@ public void onCancel(Object arguments) {
     receiver = null;
   }
 
-  private static String getNetworkType(int type) {
+  @SuppressWarnings("deprecated")
+  private String getNetworkType(int type) {
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
After discussion with @mklim, we think we should refactor this if statement out to a different method and make getNetworkType() takes the `manager` as the parameter.This way, we can keep the "deprecated" code in a single method. Do you mind to do a refactoring like this?

Reply:
No issues, I will make the changes

Reply:
Have a look, is it right?